### PR TITLE
Fix type assert from int to int64.

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

**Comment by tschottdorf** [comment]
LGTM after squashing the first commit into the second.  The commit message could mention your test update as well.

**Comment by vivekmenezes** [comment]
Changed this a bit to be safer, and addressed your comment.

File: sql/server.go [hunk 1]
```diff
@@ -139,8 +144,9 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 		return parser.DString(t), true
 	case *string:
 		return parser.DString(*t), true
+	default:
+		panic(fmt.Sprintf("Incorrect type %T", t))
```
Comment (petermattis) [inline sql/server.go]: `s/Incorrect/unexpected/g`?

Comment (vivekmenezes) [inline sql/server.go]: I'll incorporate all the feedback and fix the commented out test tonight.

**Pull request merged**